routers/common/repo.go

```diff
@@ -104,3 +106,34 @@ func ServeData(ctx *context.Context, name string, size int64, reader io.Reader)
 	_, err = io.Copy(ctx.Resp, reader)
 	return err
 }
+
+func ServeLargeFile(ctx *context.Context, name string, time time.Time, reader io.ReadSeeker) error {
+	name = strings.ReplaceAll(name, ",", " ")
+	ctx.Resp.Header().Set("Cache-Control", "public,max-age=86400")
+	st, err := typesniffer.DetectContentTypeExtFirst(name, reader)
+	if nil != err {
+		return err
+	}
+	if st.IsText() || ctx.FormBool("render") {
+		cs, err := charset.DetectEncoding(reader)
+		if err != nil {
+			log.Error("Detect raw file %s charset failed: %v, using by default utf-8", name, err)
+			cs = "utf-8"
+		}
+		// http.ServeContent has bug on detecting GBK charset
+		ctx.Resp.Header().Set("Content-Type", fmt.Sprintf("%s; charset=%s", st.Mime(), strings.ToLower(cs)))
+	} else if (st.IsImage() || st.IsPDF()) && (setting.UI.SVG.Enabled || !st.IsSvgImage()) {
+		ctx.Resp.Header().Set("Access-Control-Expose-Headers", "Content-Disposition")
+		ctx.Resp.Header().Set("Content-Disposition", fmt.Sprintf(`inline; filename="%s"`, name))
+		if st.IsSvgImage() {
+			ctx.Resp.Header().Set("Content-Security-Policy", "default-src 'none'; style-src 'unsafe-inline'; sandbox")
+			ctx.Resp.Header().Set("X-Content-Type-Options", "nosniff")
+			ctx.Resp.Header().Set("Content-Type", typesniffer.SvgMimeType)
+		}
+	} else {
+		ctx.Resp.Header().Set("Access-Control-Expose-Headers", "Content-Disposition")
+		ctx.Resp.Header().Set("Content-Disposition", fmt.Sprintf(`attachment; filename="%s"`, name))
+	}
```

Comment:
This header stuff is duplicate with `routers/common/repo.go`. Can we introduce a shared function to remove the duplicate parts?